### PR TITLE
py-pytorch-lightning: add v2.0.7

### PR DESCRIPTION
**Pull request by rfbgo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pytorch-lightning/package.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class PyPytorchLightning(PythonPackage):
 
     license("Apache-2.0")
 
+    version("2.0.7", sha256="632dec9af8036f726904e691c505d7650658ef0f4054a062c9e6a940ca09dfd8")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-pytorch-lightning/package.py]: I think you forgot to replace the sha256 after copying the directive for `2.0.0`.

I cannot confirm this sha256.  I'm getting `4e6bc1e1f7b0c69016ea2fe0616b18fa62bd3d8661c7ff02c11d317746cfc5f5`, which matches the one posted on pypi.org.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
The sha256 is incorrect.  Will defer rest of checks to @adamjstewart

File: var/spack/repos/builtin/packages/py-pytorch-lightning/package.py (hunk 1 above)
Comment (rfbgo) [reply var/spack/repos/builtin/packages/py-pytorch-lightning/package.py]: Good catch, thanks!

**Review by adamjstewart — APPROVED** [review]
I see no problems with the recipe, although I do have comments.

FYI, pytorch-lightning was renamed to lightning, and there is a new `py-lightning` package which I'm actually keeping up to date. If you still rely on the old pytorch-lightning package, that's fine, but you may want to consider switching to lightning eventually.

Why 2.0.7? There are dozens of newer versions available: https://pypi.org/project/pytorch-lightning/#history

**Comment by rfbgo** [comment]
> I see no problems with the recipe, although I do have comments.
> 
> FYI, pytorch-lightning was renamed to lightning, and there is a new `py-lightning` package which I'm actually keeping up to date. If you still rely on the old pytorch-lightning package, that's fine, but you may want to consider switching to lightning eventually.
> 
> Why 2.0.7? There are dozens of newer versions available: https://pypi.org/project/pytorch-lightning/#history

Ah, perfect, let me try swap over to py-lightning instead and we can likely abandon this

The inclusion of 2.0.7 is from a specific upstream dep that I needed to support (rather than a generic attempt to update the package)

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256.  Thanks for the fix.

**Review by bernhardkaindl — APPROVED** [review]

**Pull request merged**